File: app/src/debug/java/com/mycompany/myapp/modules/CrashReporterModule.java
package com.mycompany.myapp.modules;

import com.mycompany.myapp.monitoring.CrashReporter;
import com.mycompany.myapp.monitoring.LoggingOnlyCrashReporter;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class CrashReporterModule {
    @Singleton
    @Provides
    public CrashReporter crashReporter() {
        return new LoggingOnlyCrashReporter();
    }
}

File: app/src/debug/java/com/mycompany/myapp/modules/CrashReporterModule.kt
package com.mycompany.myapp.modules

import com.mycompany.myapp.monitoring.LoggingOnlyCrashReporter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class CrashReporterModule {
    @Singleton
    @Provides
    fun crashReporter() = LoggingOnlyCrashReporter()
}
